fda/basis.py

```diff
@@ -2189,38 +2190,51 @@ def __getitem__(self, key):
 
         if isinstance(key, int):
             return self.copy(coefficients=self.coefficients[key:key + 1])
-
         else:
             return self.copy(coefficients=self.coefficients[key])
 
+    def plus_samples(self):
+        if self.nsamples == 1:
+            return self
+        return self[0] + (self[1:].plus_samples())
+
     def __add__(self, other):
         """Addition for FDataBasis object."""
+        if isinstance(other, FDataBasis):
+            if self.basis != other.basis:
+                raise NotImplementedError
+            else:
+                return FDataBasis(self.basis.copy(),
+                                  self.coefficients + other.coefficients)
 
-        raise NotImplementedError
+        coefs = self.coefficients.copy()
+        coefs[:, 0] = self.coefficients[:, 0] + numpy.array(other)
```

Comment:
We should check if `other` can be used as an `ndarray` and return `NotImplemented` otherwise.

Reply:
How can I check if an object is numpy arrayable?

Reply:
Try the conversion and catch the possible exception.